src/main/java/com/ericsson/esj/JenkinsManager.java

```diff
@@ -0,0 +1,259 @@
+        }
+
+        String endpoint = "/job/" + jobName + "/build";
+
+        httpRequest.setJenkinsBaseUrl(jenkinsBaseUrl).addHeader("Authorization", "Basic " + encoding)
+                .addHeader("Content-type", MediaType.APPLICATION_JSON).addParam("token", job_token)
+                .setEndpoint(endpoint);
+
+        ResponseEntity response = httpRequest.performRequest();
+        success = response.getStatusCode() == HttpStatus.SC_CREATED;
+
+        if(!success) {
+            LOGGER.error("Failed to trigger a jenkins job. Status code: " + response.getStatusCodeValue() + ". Reason: " + response.getBody());
+        }
+
+        return success;
+    }
+
+    /**
+     * This function recieves a jenkins job name and returns whether the job has
+     * been triggered at least once
+     *
+     * @param jobName
+     * @return
+     * @throws URISyntaxException
+     */
+    public boolean jobHasBeenTriggered(String jobName) throws URISyntaxException {
+        boolean isTriggered = false;
+        HttpRequest httpRequest = new HttpRequest(HttpMethod.GET);
+        if (jobName == null) {
+            jobName = DEFAULT_JOB_NAME;
+        }
+
+        String endpoint = "/job/" + jobName + "/1/api/json";
+
+        httpRequest.setJenkinsBaseUrl(jenkinsBaseUrl).addHeader("Authorization", "Basic " + encoding)
+                .addHeader("Content-type", MediaType.APPLICATION_JSON)
+                .setEndpoint(endpoint);
+
+        ResponseEntity response = httpRequest.performRequest();
+        isTriggered = response.getStatusCode() == HttpStatus.SC_OK;
+
+        if(!isTriggered) {
+            LOGGER.error("Failed to check if job was triggered. Status code: " + response.getStatusCodeValue() + ". Reason: " + response.getBody());
+        }
+
+        return isTriggered;
+    }
+
+    /**
+     *
+     * This function recieves a jenkins job name and deletes that job from the
+     * jenkins system
+     *
+     * @param jobName
+     * @return
+     * @throws URISyntaxException
+     */
+    public boolean deleteJob(String jobName) throws URISyntaxException {
+        boolean isDeleted = false;
+        HttpRequest httpRequest = new HttpRequest(HttpMethod.POST);
+        if (jobName == null) {
+            jobName = DEFAULT_JOB_NAME;
+        }
+
+        String endpoint = "/job/" + jobName + "/doDelete";
+
+        httpRequest.setJenkinsBaseUrl(jenkinsBaseUrl).addHeader("Authorization", "Basic " + encoding)
+                .addHeader("Content-type", MediaType.APPLICATION_JSON).addHeader("Jenkins-Crumb", crumb)
+                .setEndpoint(endpoint);
+
+        ResponseEntity response = httpRequest.performRequest();
+        isDeleted = response.getStatusCode() == HttpStatus.SC_MOVED_TEMPORARILY;
+
+        if(!isDeleted) {
+            LOGGER.error("Failed to delete jenkins job. Status code: " + response.getStatusCodeValue() + ". Reason: " + response.getBody());
+        }
+
+        return isDeleted;
+    }
+
```

Comment:
Add java-doc, we said we should add it to every method, except if it's super obvious :)